### ISSUE TITLE
Support FORCE_COLOR / NO_COLOR env vars

### ISSUE THREAD

**Issue by mjpieters** [crate-ci/typos#1141]
Please support the [`FORCE_COLOR`](https://force-color.org/) and [`NO_COLOR`](https://no-color.org/) environment variables as alternative to the`--color` CLI switch.

The start-up behaviour then would be:

1. `--color` set: use this configuration.
2. `NO_COLOR` set: treat as `--color never`
3. `FORCE_COLOR` set: treat as `--color always`
4. use the default `--color auto`. 

I see typos uses `anstyle`, like `uv`, so this is a virtual duplicate of astral-sh/uv#3955, with `NO_COLOR` already working, and `CLICOLOR_FORCE` having the same effect as `FORCE_COLOR`.

**Comment by epage** [comment]
This is blocked on rust-cli/anstyle#192 as I'm not going to move forward like uv did in astral-sh/uv#3979 due to all of the open question related to `FORCE_COLOR` (see donatj/force-color.org#30).  However, this shouldn't be blocking because `CLICOLOR_FORCE` exists.

**Comment by mjpieters** [comment]
My PR interprets `FORCE_COLOR` the same way `CLICOLOR_FORCE` is interpreted: as non-empty.
